Cython/Compiler/Dataclass.py

```diff
@@ -221,6 +222,8 @@ def handle_cclass_dataclass(node, dataclass_args, analyse_decs_transform):
 
     fields = process_class_get_fields(node)
 
+    kw_only = kwargs.pop("kw_only")
```

Comment:
but `_DataclassParams` doesn't have a `kw_only` argument (neither in regular data class)

Reply:
Hmm. Ok. Sorry, I didn't read the code completely. So, you were only using `kwargs` for the keyword parsing here and it's otherwise passed straight into dataclass params. I reverted my change and added a comment on why it is done this way. It's not obvious from the code as such.